src/detect-engine-port.c

```diff
@@ -1345,15 +1351,12 @@ DetectPort *PortParse(const char *str)
  * \retval 1 if port is in the valid range
  * \retval 0 if invalid
  */
-int DetectPortIsValidRange(char *port)
+static int DetectPortIsValidRange(char *port)
 {
-    char *end;
-    long r = strtol(port, &end, 10);
-
-    if(*end == 0 && r >= 0 && r <= 65535)
-        return 1;
-    else
-        return 0;
+    uint16_t val;
+    if (StringParseUint16(&val, 10, 0, (const char *)port) < 0)
+        return -1;
+    return val;
```

Comment:
The return values should be either `1` or `0` .. perhaps this static function could return a bool?

Reply:
I think the comments of the function are wrong and need to be fixed

Reply:
Ok ... suggest that
1. The function return a `bool` indicating success/failure
2. The function signature change to `char *port, uint16_t *port_val)`

Reply:
Point 1 done.
Point 2: Could you please tell why is it better to pass port value in this function than where it is now?

Reply:
I think Jeff's suggestion is about differentiating the return code : error or failure, and the port value
Using an `int` for both at the same time may not be a good coding pattern

Reply:
I see. Thanks @catenacyber !